apps/fishing-map/features/resources/resources.slice.ts

```diff
@@ -14,12 +16,17 @@ export {
 } from '@globalfishingwatch/dataviews-client'
 
 export const selectThinningConfig = createSelector(
-  [(state) => isGuestUser(state), selectDebugOptions],
-  (guestUser, { thinning }) => {
+  [(state) => isGuestUser(state), selectDebugOptions, selectUrlMapZoomQuery],
+  (guestUser, { thinning }, currentZoom) => {
     if (!thinning) return null
-    const thinningConfig = guestUser
-      ? THINNING_LEVELS[ThinningLevels.Aggressive]
-      : THINNING_LEVELS[ThinningLevels.Default]
+    let thinningConfig: ThinningConfig
+    for (let i = 0; i < THINNING_LEVEL_ZOOMS.length; i++) {
+      const zoom = THINNING_LEVEL_ZOOMS[i]
+      if (currentZoom < zoom) break
+      thinningConfig = THINNING_LEVEL_BY_ZOOM[zoom][guestUser ? 'guest' : 'user']
```

Comment:
I think this could be a good place of deciding what thining zoom level choose while switching zoom levels.

We could use `resolveResourcesFromDatasetConfigs` to know in advance which resourceUrl string is going to be generated and check the resources status with:

`import { selectResources } from './resources.slice'` so then we can:

- use it if already loaded
- if not loaded we can try with the lower zoom level in case it is available


this also would allow us to avoid requesting lower zoom levels when the highest one was already loaded (for example a workspace in zoom level 12 by default won't need the zoom 4 thinning config)